### PR TITLE
Prereq indicators

### LINKED ISSUE #288: prereq indicators in class info card
It would be helpful to have an indicator of what we consider fulfilled or not (like little checks or xes)

### PR DESCRIPTION
**Pull request by georgiashay**
Closes #288 

- moves some requirement logic into a shared mixin
- lightens requirements which are fulfilled in prereqs expansion-req (so it looks like fireroad does it)
- moves around some things to get the class info card/expansion req to update properly

### CODE REVIEW

**Comment by npfoss** [comment]
update: want to add green check marks in the top right also

File: src/components/ClassInfo.vue [hunk 1]
```diff
@@ -263,6 +264,16 @@ export default {
           return 0;
         }
       });
+    },
+    firstAppearance: function () {
+      const currentSubjects = this.$store.state.roads[this.$store.state.activeRoad].contents.selectedSubjects;
+      const currentID = this.currentSubject.subject_id;
+
+      const subjectInSemesters = currentSubjects.map((semesterSubjects) => {
+        return semesterSubjects.map((subj) => subj.id).indexOf(currentID) >= 0;
```
[the remark this answers is not in the document]
Comment (georgiashay) [reply src/components/ClassInfo.vue]: browser support for contains not great

File: src/components/ExpansionReqs.vue [hunk 2]
```diff
@@ -80,15 +81,29 @@ export default {
...
   watch: {
-    requirement: function (newReq, oldReq) {
+    reqID: function (newReq, oldReq) {
       this.open = false;
     }
   },
   methods: {
     clickSubject: function (subj, scroller) {
+      this.scrollerClicked = scroller;
+      this.subjectClicked = subj;
```
[the remark this answers is not in the document]
Comment (georgiashay) [reply src/components/ExpansionReqs.vue]: Yes.  For the double scrollers we need to know which scroller was clicked, and we need to know which subject is clicked within the scroller.

[the remark this answers is not in the document]
Comment (npfoss) [reply src/components/ExpansionReqs.vue]: ah, gotcha. didn't realized how the specific scroller was being used before

**Pull request merged**